components/ReceivedAssets/ReceivedAssets.tsx

```diff
@@ -0,0 +1,77 @@
+import { useEffect, useState } from 'react';
+import { ethers } from 'ethers';
+
+import useEthersProvider from '../../hooks/useEthersProvider';
+import fetchReceivedAssets from '../../utils/fetchReceivedAssets';
+import useWeb3Provider from '../../hooks/useWeb3Provider';
+import isLSP7orLSP8 from '../../utils/isLSP7orLSP8';
+import { LSPType } from '../../interfaces/lsps';
+import LSP7Table from '../../components/LSP7Table/LSP7Table';
+import LSP8Table from '../../components/LSP8Table/LSP8Table';
+
+interface IProps {
+  isUniversalProfile: boolean;
+  ownerAddress: string;
+}
+
+const ReceivedAssets: React.FC<IProps> = ({
+  isUniversalProfile,
+  ownerAddress,
+}) => {
+  const [lsp7Addresses, setLsp7Addresses] = useState<string[]>([]);
+  const [lsp8Addresses, setLsp8Addresses] = useState<string[]>([]);
+
+  const provider = useEthersProvider() as ethers.providers.BaseProvider;
+
+  //ERC725 does not support ethers provider
+  const web3Provider = useWeb3Provider();
+
+  const fetchAssets = async (address: string) => {
+    //fetch all received assets for specific up address
+    const receivedAssets = await fetchReceivedAssets(address, web3Provider);
+
+    const lsp7AddressesTemp: string[] = [];
+    const lsp8AddressesTemp: string[] = [];
+
+    //fetch the different assets types
+    await Promise.all(
+      receivedAssets.map(async (assetAddress) => {
+        const assetType = await isLSP7orLSP8(assetAddress, provider);
+        switch (assetType) {
+          case LSPType.LSP7:
+            lsp7AddressesTemp.push(assetAddress);
+            break;
+          case LSPType.LSP8:
+            lsp8AddressesTemp.push(assetAddress);
+            break;
+          default:
+            break;
+        }
+      }),
+    );
+    setLsp7Addresses(lsp7AddressesTemp);
+    setLsp8Addresses(lsp8AddressesTemp);
```

Comment:
I see this piece of code a lot in the code. Here, in CreatedAssets, etc. I think it is time to move this into a function that looks liks this:

input: string[] a list of assets addresses
output: 
```
{
    lsp7Addresses: string[] list of lsp7 addresses,
    lsp8Addresses: string[] list of lsp8 addresses
}
```

So you can reuse that function everywhere you do this try / catch.

If you use the same thing more than 2 times, then use the same function.

![image](https://user-images.githubusercontent.com/477945/187885368-206a81ad-bbbd-4387-ad91-65ce127481a5.png)